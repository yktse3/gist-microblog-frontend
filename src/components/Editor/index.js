import React from 'react';
import {
  Title,
  Content,
  SubmitBtn,
} from './styles';
import {
  Card,
} from '../Article/styles';

const Editor = () => (
  <Card>
    <Title
      placeholder="Title"
    />
    <Content
      placeholder="Content"
    />
    <SubmitBtn>Submit</SubmitBtn>
  </Card>
);

export default Editor;
