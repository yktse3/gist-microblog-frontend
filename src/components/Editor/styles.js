import styled from 'styled-components';

export const Title = styled.input`
  height: 33px;
  font-size: 20px;
  margin-bottom: 20px;
  outline-color: #d69d56;
`;

export const Content = styled.textarea`
  resize: vertical;
  font-size: 20px;
  height: 160px;
  margin-bottom: 15px;
  outline-color: #d69d56;
`;

export const SubmitBtn = styled.button`
  width: fit-content;
  border: 1px solid #2c252e;
  color: #2c252e;
  background: #d69d56;
  padding: 10px 20px;
  align-self: flex-end;
  outline: none;
`;
