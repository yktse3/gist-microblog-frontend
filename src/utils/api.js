const CLIENT_ID = 'b77a50cf0d2a3029972c';
const CLIENT_SECRET = 'f29284e62d8c8bdd0fba0b304b7c5c243eec0d88';
const HEADER_ACCEPT_JSON = 'application/json';
const HEADER_ACCEPT_GITHUB_V3 = 'application/vnd.github.v3+json';

const defaultConfig = {
  accessToken: null,
};

class API {
  constructor(config = defaultConfig) {
    this._config = config;
  }

  changeAccessToken = (accessToken) => {
    this._config.accessToken = accessToken;
  }

  getAccessToken = async ({ code }) => {
    const url = 'http://localhost:3100/login';
    const body = {
      client_id: CLIENT_ID,
      client_secret: CLIENT_SECRET,
      code,
    };

    const config = {
      method: 'POST',
      headers: {
        Accept: HEADER_ACCEPT_JSON,
        'Content-Type': HEADER_ACCEPT_JSON,
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify(body),
    };

    try {
      const result = await fetch(url, config);
      const response = await result.json();
      return response;
    } catch (e) {
      return Promise.reject(e);
    }
  };
}

export default new API();
