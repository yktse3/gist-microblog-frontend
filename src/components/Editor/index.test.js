import React from 'react';
import renderer from 'react-test-renderer';
import { noop } from 'utils/helpers';
import { mount } from 'enzyme';
import Editor from './index';

describe('Editor', () => {
  it('render successfully when enabled', () => {
    const component = renderer.create(
      <Editor
        onSubmit={noop}
        disabled
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render successfully when disabled', () => {
    const component = renderer.create(
      <Editor
        onSubmit={noop}
        disabled={false}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('change state when typing in content box', () => {
    const wrapper = mount(
      <Editor
        onSubmit={noop}
        disabled={false}
      />,
    );
    wrapper
      .find('textarea')
      .first()
      .simulate('change', {
        target: {
          name: 'content',
          value: 'new article content',
        },
      });
    expect(wrapper.state('content')).toEqual('new article content');
  });

  it('change state when typing in title box', () => {
    const wrapper = mount(
      <Editor
        onSubmit={noop}
        disabled={false}
      />,
    );
    wrapper
      .find('input')
      .first()
      .simulate('change', {
        target: {
          name: 'title',
          value: 'new article title',
        },
      });
    expect(wrapper.state('title')).toEqual('new article title');
  });

  it('calls onSubmit with correct parameters', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(
      <Editor
        onSubmit={onSubmit}
        disabled={false}
      />,
    );

    const state = {
      title: 'new article title',
      content: 'new article content',
    };
    wrapper.setState(state);

    wrapper
      .find('button')
      .first()
      .simulate('click');

    expect(onSubmit).toBeCalledWith(state);

    expect(wrapper.state()).toEqual({
      title: '',
      content: '',
    });
  });
});
