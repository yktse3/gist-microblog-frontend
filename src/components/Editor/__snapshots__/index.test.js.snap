// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor render successfully when disabled 1`] = `
<div
  className="sc-EHOje jiRcjx"
>
  <input
    autoComplete="off"
    className="sc-bdVaJa dvXOK"
    disabled={false}
    name="title"
    onChange={[Function]}
    placeholder="Title"
    value=""
  />
  <textarea
    autoComplete="off"
    className="sc-bwzfXH jGzomT"
    disabled={false}
    name="content"
    onChange={[Function]}
    placeholder="Content"
    value=""
  />
  <button
    className="sc-htpNat eTyvdL"
    disabled={false}
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`Editor render successfully when enabled 1`] = `
<div
  className="sc-EHOje jiRcjx"
>
  <div
    className="sc-ifAKCX TUjuP"
  />
  <div
    className="sc-bxivhb jYnjvO"
  >
    <div>
      <svg
        height="100"
        viewBox="0 0 38 38"
        width="100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#d69d56"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#d69d56"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#d69d56"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="#d69d56"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r="1"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
  <input
    autoComplete="off"
    className="sc-bdVaJa dvXOK"
    disabled={true}
    name="title"
    onChange={[Function]}
    placeholder="Title"
    value=""
  />
  <textarea
    autoComplete="off"
    className="sc-bwzfXH jGzomT"
    disabled={true}
    name="content"
    onChange={[Function]}
    placeholder="Content"
    value=""
  />
  <button
    className="sc-htpNat eTyvdL"
    disabled={true}
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
